src/main/drivers/dshot.c

```diff
@@ -0,0 +1,128 @@
+/*
+ * This file is part of Cleanflight and Betaflight.
+ *
+ * Cleanflight and Betaflight are free software. You can redistribute
+ * this software and/or modify this software under the terms of the
+ * GNU General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option)
+ * any later version.
+ *
+ * Cleanflight and Betaflight are distributed in the hope that they
+ * will be useful, but WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Author: jflyper
+ */
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <math.h>
+
+#include "platform.h"
+
+#ifdef USE_DSHOT
+
+#include "common/maths.h"
+#include "common/time.h"
+
+#include "config/feature.h"
+
+#include "drivers/motor.h"
+#include "drivers/timer.h"
+
+#include "drivers/dshot.h"
+#include "drivers/dshot_dpwm.h" // for motorDmaOutput_t, should be gone
+#include "drivers/dshot_command.h"
+#include "drivers/pwm_output.h" // for PWM_TYPE_* and others
+
+#include "fc/rc_controls.h" // for flight3DConfig_t
+
+#include "pg/motor.h"
+
+#include "rx/rx.h"
+
+void dshotInitEndpoints(float outputLimit, float *outputLow, float *outputHigh, float *disarm, float *deadbandMotor3dHigh, float *deadbandMotor3dLow) {
```

Comment:
Maybe consolidate this into an `endpoints` struct, this won't carry any runtime or binary size overhead, just the advantages.

Reply:
Sensible.

Reply:
Hmm... Tried in such a was that these parameters (variables actually) are stored in `struct endPoint_s`, but it made other numerous references to them as structure member access which made the code bit unclear.

I'd leave it as is, since there's not much overhead as end point initialization is called once per run.

Reply:
I'm mostly concerned that functions with lots of parameters look heavy.
Leaving it up to you